mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -947,6 +947,16 @@ void CMomentumPlayer::OnZoneEnter(CTriggerZone *pTrigger)
             SetCurrentZoneTrigger(pStartTrigger);
             SetCurrentProgressTrigger(pStartTrigger);
 
+            if (g_pGameModeSystem->GameModeIs(GAMEMODE_RJ))
+            {
+                // Don't limit speed in rocketjump mode,
+                // reset timer on zone enter and start on zone leave.
+                g_pMomentumTimer->Reset(this);
```

Comment:
With no check of the player being on the ground and not in noclip mode (having MOVETYPE_WALK), players will be able to restart their timer by just touching the trigger on its top-most edge, for example. It's very important we do not allow this.

I also think this can be better. I believe game mode should have some sort of override to the limit speed type and if it should start on jump. The order of setting `m_iSpeedLimitType` and `m_bShouldLimitSpeed` going Game Mode -> Map-wide property (#233 ) -> Individual Track Property (also #233 ).

This work I'm describing may be best for another branch, but the problem remains still with the Reset.